wire-runtime/src/main/java/com/squareup/wire/WireAdapter.java

```diff
@@ -347,6 +334,27 @@ public String toString(E value) {
     }
   };
 
+  private static final Map<WireType, WireAdapter<?>> TYPE_TO_ADAPTER;
+  static {
+    Map<WireType, WireAdapter<?>> map = new LinkedHashMap<WireType, WireAdapter<?>>();
+    map.put(WireType.BOOL, WireAdapter.BOOL);
```

Comment:
... in which case, we could drop the 15 constants from `WireAdapter` that 100% correspond to the 15 constants in `WireType`.